tests/hil/scripts/pytest-hil/espboard.py

```diff
@@ -0,0 +1,19 @@
+import esptool
+
+from board import Board
+
+class ESPBoard(Board):
+    def __init__(self, app_offset):
+        self.app_offset = str(app_offset)
+
+    def program(self, fw_image):
+        cmd = [
+            "--port", self.port,
+            "--baud", "460800",
+            "write_flash",
+            "-e",
+            self.app_offset,
+            fw_image,
+        ]
+
+        esptool.main(cmd)
```

Comment:
Ugly, but this is what Espressif recommends: https://docs.espressif.com/projects/esptool/en/latest/esp32/esptool/scripting.html